codegen/projections/white_label/lib/white_label/event_stream.rb

```diff
@@ -96,15 +82,38 @@ def on_unknown_event(&block)
 
       def parse_event(type, message)
         case type
-        when 'initial-response' then Parsers::EventStream::StartEventStreamInitialResponse.parse(message)
-        when 'SimpleEvent' then Types::Events::SimpleEvent.new(Parsers::EventStream::SimpleEvent.parse(message))
-        when 'NestedEvent' then Types::Events::NestedEvent.new(Parsers::EventStream::NestedEvent.parse(message))
-        when 'ExplicitPayloadEvent' then Types::Events::ExplicitPayloadEvent.new(Parsers::EventStream::ExplicitPayloadEvent.parse(message))
-        when 'ServerErrorEvent' then Types::Events::ServerErrorEvent.new(Parsers::EventStream::ServerErrorEvent.parse(message))
+        when 'initial-response'
+          Parsers::EventStream::StartEventStreamInitialResponse.parse(message)
+        when 'SimpleEvent'
+          Types::Events::SimpleEvent.new(Parsers::EventStream::SimpleEvent.parse(message))
+        when 'NestedEvent'
+          Types::Events::NestedEvent.new(Parsers::EventStream::NestedEvent.parse(message))
+        when 'ExplicitPayloadEvent'
+          Types::Events::ExplicitPayloadEvent.new(Parsers::EventStream::ExplicitPayloadEvent.parse(message))
+        when 'ServerErrorEvent'
+          Types::Events::ServerErrorEvent.new(Parsers::EventStream::ServerErrorEvent.parse(message))
         else
           Types::Events::Unknown.new(name: type || 'unknown', value: message)
         end
       end
+
+      def parse_exception_event(type, message)
+        case type
+        when 'ServerErrorEvent'
+          data = Parsers::EventStream::ServerErrorEvent.parse(message)
+          Errors::ServerErrorEvent.new(data: data, error_code: 'WhiteLabel::Types::Events::ServerErrorEvent')
+        else
+          data = Types::Events::Unknown.new(name: type || 'unknown', value: message)
```

Comment:
Maybe 'unknown' should be baked into the type as a default name? Maybe error code as well?

Reply:
Yeah good call, I think that makes sense.